### PR TITLE
migrating component environment

### PR DESCRIPTION
**Pull request by jayesh-tanna**
-migrating component env to azureml-assets

### CODE REVIEW

File: assets/component/asset.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: component
```
Comment (vizhur) [inline assets/component/asset.yaml:1]: folder should probably be named environment?

File: assets/component/context/conda_dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: component
+channels:
+- conda-forge
+dependencies:
+- python=3.9
+- pip=21.3.1
```
Comment (vizhur) [inline assets/component/context/conda_dependencies.yaml:6]: probably worth to bump pip to something more recent

File: assets/component/context/conda_dependencies.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+- pip:
+  - azure-ml-component=={{latest-pypi-version}}
+  - azureml-defaults=={{latest-pypi-version}}
+  - requests=={{latest-pypi-version:pre}}
```
Comment (vizhur) [inline assets/component/context/conda_dependencies.yaml:10]: please remove cryptography, requests and urllib3

File: assets/component/asset.yaml (hunk 1 above)
Comment (jayesh-tanna) [reply assets/component/asset.yaml:1]: agree and moved under environments directory.

File: assets/component/context/conda_dependencies.yaml (hunk 2 above)
Comment (jayesh-tanna) [reply assets/component/context/conda_dependencies.yaml:6]: upgraded to 24.0 version instead of 21.3.1 after compatibility check

**Review by vizhur — APPROVED** [review]

File: assets/component/asset.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+version: auto
```
[the remark this answers is not in the document]
Comment (jayesh-tanna) [reply assets/component/asset.yaml]: can you please help here to make changes from backend side.

File: assets/designer/environments/component/context/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04:{{latest-image-tag}}
```
Comment (vizhur) [inline assets/designer/environments/component/context/Dockerfile:1]: use ubuntu22.04

Comment (jayesh-tanna) [reply assets/designer/environments/component/context/Dockerfile]: upgraded to ubuntu22.04

**Review by achauhan-scc — APPROVED** [review]

**Pull request merged**